Task06_Yoav_Uzan/src/main.cpp

```diff
@@ -0,0 +1,153 @@
+#include <functional>
+#include <vector>
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <string>
+#include "Location.cpp"
+#include <fstream>
+#include <filesystem>
+#include "SortByMap.cpp"
+
+/**
+  brief Calculate the Euclidean distance between two locations.
+  param loc1 The first location.
+  param loc2 The second location.
+  return The Euclidean distance between the two locations.
+ */
+double L2(const Location& loc1, const Location& loc2)
+{
```

Comment:
there is std::function in the STL library that you can use and not a pointer to function